### PR TITLE
[1LP][RFR] Fix for typo in sprout's poke_trackerbot

### PR DESCRIPTION
**Pull request by izapolsk**
This PR fixes some issues in sprout tasks, finally removes container property, adds template_type to view and so on.
It should be last patch which adds support of openshift provider in sprout

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: sprout/appliances/models.py [hunk 1]
```diff
@@ -1171,9 +1171,9 @@ def create(cls, owner, group, version=None, date=None, provider=None, num_applia
             time_leased=60, preconfigured=True, yum_update=False, container=False, ram=None,
             cpu=None, provider_type=None):
         if container:
-            container_q = ~Q(container=None) & ~Q(provider_type='openshift')
+            container_q = ~Q(container=None) & ~Q(provider__provider_type='openshift')
         else:
-            container_q = Q(container=None)
+            container_q = Q(container=None) | Q(provider__provider_type='openshift')
```
Comment (mfalesni) [inline sprout/appliances/models.py]: Shouldn't these two branches be swapped? I am confused a bit.

Comment (izapolsk) [reply sprout/appliances/models.py]: When we discussed this point last time, you mentioned that containers and openshift appliances are different stuff. So, it got me thinking that openshift appliances have to be attributed to regular appliances.

Comment (psav) [reply sprout/appliances/models.py]: @mfalesni please provide resolution to this comment

Comment (mfalesni) [reply sprout/appliances/models.py]: @psav we were discussing this on IRC

File: sprout/appliances/views.py [hunk 2]
```diff
@@ -680,7 +688,7 @@ def request_pool(request):
             # Container is preconfigured only
             # We need to do this as the disabled checkbox for Preconfigured seems to not return
             # the proper value.
-            preconfigured = True
+            preconfigured = False if provider_type == 'openshift' else True
```
Comment (mfalesni) [inline sprout/appliances/views.py]:
```python
preconfigured = provider_type != 'openshift'
```

File: sprout/appliances/tasks.py [hunk 3]
```diff
@@ -1968,8 +1971,8 @@ def process_docker_images_from_url_group(self, group_id, version, docker_version
             continue
         try:
             Template.objects.get(
-                ~Q(container=None), template_group=group, provider=provider, version=version,
-                date=date, preconfigured=True)
+                ~Q(container=None) & ~Q(provider__provider_type='openshift'), template_group=group,
+                provider=provider, version=version, date=date, preconfigured=True)
```
Comment (mfalesni) [inline sprout/appliances/tasks.py]: I think this should just plainly skip and not execute at all for openshift.

Comment (izapolsk) [reply sprout/appliances/tasks.py]: you're right. I even added such check several rows before

File: sprout/appliances/models.py [hunk 4]
```diff
@@ -1171,7 +1174,10 @@ def create(cls, owner, group, version=None, date=None, provider=None, num_applia
             time_leased=60, preconfigured=True, yum_update=False, container=False, ram=None,
             cpu=None, provider_type=None):
         if container:
-            container_q = ~Q(container=None) & ~Q(provider_type='openshift')
+            if provider_type != 'openshift':
```
Comment (RonnyPfannschmidt) [inline sprout/appliances/models.py]: both look exactly the same i beleive thats not intended

Comment (RonnyPfannschmidt) [reply sprout/appliances/models.py]: forget it, i did not see the `~` - sorry

File: sprout/appliances/views.py [hunk 5]
```diff
@@ -209,7 +208,7 @@ def versions_for_group(request):
                 for version in Template.get_versions(
                     container_q,
                     template_group=group, ready=True, usable=True, exists=True,
-                    preconfigured=preconfigured, provider__working=True, provider__disabled=False,
```
Comment (mfalesni) [inline sprout/appliances/views.py]: Hmm, you cannot remove the preconfigured boolean from these queries, sometimes preconfigured appliances are not available and pure ones are, which would make confusion over failing requests.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by izapolsk** [comment]
requires #6769

**Pull request merged**